### ISSUE TITLE
why reward is penalized in line 140/141 in modules.py?

### ISSUE THREAD

**Issue by zonglunli7515** [Thartvigsen/StopAndHop#3]
Hi,

in line 140 in modules.py, if log_pi_stop is negative, then a positive reward would contribute to the loss function. What is the logic behind it? 

I am new to reinforcement learning and pytorch, apologies if this is quite obvious.

Thanks.

Zonglun

**Comment by Thartvigsen** [comment]
Minimizing the negative log probability maximizes the probability. So if the reward's positive, then the log probability of the chosen action will be minimized, increasing the probability of the actions that led to that reward!

**Comment by zonglunli7515** [comment]
> Minimizing the negative log probability maximizes the probability. So if the reward's positive, then the log probability of the chosen action will be minimized, increasing the probability of the actions that led to that reward!

Many thanks for your speedy response. Initially I thought reward is sth more rewarding, and therefore should help reduce the total loss. Minimizing the negative log makes perfect sense.
